src/main/java/pl/mmieczak/springbootexercise_22_1/UserController.java

```diff
@@ -0,0 +1,48 @@
+package pl.mmieczak.springbootexercise_22_1;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+public class UserController {
+
+    private pl.mmieczak.springbootexercise_22_1.UserRepository userRepository;
+
+    public UserController(pl.mmieczak.springbootexercise_22_1.UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    //@PostMapping("/add")
+    @RequestMapping("/add")
+    public String add(@RequestParam String name, @RequestParam(defaultValue = "", required = false) String surname, @RequestParam(defaultValue = "-1", required = true) String age) {
+        int ageToInt = Integer.parseInt(age);
```

Comment:
To mogłeś zamianić na powiedzenie springowi że potrzebuejsz Integera. Wystarczyło przy `RequestMapping` zmienić typ ze Stringa na Integera. Spring przeprowadza wtedy automatyczną konwesję typów.

Reply:
Faktycznie :)